nixos/lib/make-zfs-image.nix

```diff
@@ -219,23 +219,12 @@ let
       nixpkgs-fmt $out
     '';
 
-  mergedConfig =
-    if configFile == null
-    then fileSystemsCfgFile
-    else
-      pkgs.runCommand "configuration.nix" {
-        buildInputs = with pkgs; [ nixpkgs-fmt ];
-      }
-        ''
-          (
-            echo '{ imports = ['
-            printf "(%s)\n" "$(cat ${fileSystemsCfgFile})";
-            printf "(%s)\n" "$(cat ${configFile})";
-            echo ']; }'
-          ) > $out
-
-          nixpkgs-fmt $out
-        '';
+  mergedConfig = pkgs.mergeNixOSModules {
```

Comment:
Unless I'm missing something, these two don't seem to be entirely identical. Shouldn't `moduleFiles` be something like `[ fileSystemsCfgFile ] ++ lib.optional (configFile != null) configFile`?

Reply:
They're not literally identical but should be functionally identical. We could go back to the more complex code which produces a simpler module file if that is useful, but I was thinking it'd be nice to have simpler code / more predictable behavior by always doing it.

Reply:
I was more wondering why you always include the `configFile`, which looking at the pre-change snippet appears to be nullable, whereas `fileSystemsCfgFile` appears to be always used in the snippet, yet only optionally in this one (it seems like it's never null, though).

To me, the current form feels identical to just `moduleFiles = [ configFile fileSystemsCfgFile ];`

Reply:
(Basically, I think you just switched the names around on accident, unless I misunderstood something.)

Reply:
OH! Thanks! Fixed.